### PR TITLE
Fix parameter-clone

### PR DESCRIPTION
**Pull request by benjello**
#### New features

Fix Parameters.clone() to yield true clones

### CODE REVIEW

File: openfisca_core/parameters.py [hunk 1]
```diff
@@ -191,6 +192,18 @@ def __eq__(self, other):
...
 
+    def clone(self):
+        new = empty_clone(self)
+        new_dict = new.__dict__
+
+        for key, value in self.__dict__.items():
+            if key not in ['values_list', 'metadata']:
+                new_dict[key] = value
+
+        new.metadata = copy.deepcopy(self.metadata)
```
Comment (fpagnoux) [inline openfisca_core/parameters.py]: > metadata is a basic Python `dict`, so we can rely on the built-in deepcopy

File: openfisca_core/parameters.py [hunk 2]
```diff
@@ -191,6 +192,18 @@ def __eq__(self, other):
...
+        new.values_list = [parameter_at_instant.clone() for parameter_at_instant in self.values_list]
```
Comment (fpagnoux) [inline openfisca_core/parameters.py]: > Using `clone` here since `parameter_at_instant ` is a custom class

File: openfisca_core/parameters.py [hunk 3]
```diff
@@ -191,6 +192,14 @@ def __eq__(self, other):
     def __call__(self, instant):
         return self.get_at_instant(instant)
 
+    def clone(self):
+        clone = empty_clone(self)
+        clone.__dict__ = self.__dict__.copy()
+
+        clone.metadata = copy.deepcopy(self.metadata)
```
Comment (fpagnoux) [inline openfisca_core/parameters.py]: > metadata is a basic Python `dict`, so we can rely on the built-in `deepcopy`

File: openfisca_core/parameters.py [hunk 4]
```diff
@@ -191,6 +192,14 @@ def __eq__(self, other):
...
+        clone.values_list = [parameter_at_instant.clone() for parameter_at_instant in self.values_list]
```
Comment (fpagnoux) [inline openfisca_core/parameters.py]: > `parameter_at_instant ` is a custom class, so using `clone`

File: tests/core/parameter_validation/test_parameter_clone.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
```
Comment (fpagnoux) [inline tests/core/parameter_validation/test_parameter_clone.py]: > I'd probably move this test outside of the `parameter_validation` folder.

Comment (benjello) [reply tests/core/parameter_validation/test_parameter_clone.py]: Donc

**Comment by benjello** [comment]
@Morendil : could we go this way for now but we wight adopt a solution along the lines you suggested on openfisca-core slack ?

**Comment by benjello** [comment]
cc @maukoquiroga

**Pull request merged**